### ISSUE TITLE
Workaround for IG `/confirms` bug

### ISSUE THREAD

**Issue by bug-or-feature** [ig-python/trading-ig#320]
There is a bug in IG's implementation of their

```
/confirms/{dealReference}
```

endpoint, when you close a position. The method in this project is `IGService.fetch_deal_by_deal_reference()`. This is the request you make after opening/updating/closing a position or working order. It responds with the status of your request, and a unique ID for the transaction, if it was successful. 

The bug is this: when closing a position, `fetch_deal_by_deal_reference()` responds with the ID of the **opening** transaction, not closing. There is a discussion about it on the IG community forum [here](https://community.ig.com/forums/topic/23128-how-does-the-relation-between-dealid-and-dealreference-work-in-the-different-api-endpoints/).

This a real pain. Especially as IG don't even acknowledge that there is anything wrong. It makes it hard to associate opening and closing trades, and therefore get P&L information.

There is a workaround, as outlined in the forum chat. I propose to add built-in support for the workaround in the code. Optional of course.

The same bug exists whether you request the `confirms` information via the REST or Streaming API

**Comment by psts** [comment]
Thanks for the workaround. Will give it a try today since I have been battling with this bug and took a while for me to figure out what was wrong.

This only seem to handle if you close a position manually. Do you know of a way to fix it when a stop is hit and result comes back through the stream?

**Comment by bug-or-feature** [comment]
I abandoned my attempt to add built in support for the workaround into the code - it was too complicated. So you're left with the workaround, as outlined in the post on the community forum. That is valid whether you're getting trade updates from REST or Streaming

**Comment by psts** [comment]
I understand. It's a mess!

There's a few adaptions I'll do since I'm using CFD's and the search filter doesn't match completely, but at least I'm on the right track. Thanks!

Btw, in the workaround, I think you have a typo:

for i in range(5):
            df = self.fetch_account_activity(
                from_date=deal_date - timedelta(seconds=1),
                to_date=deal_date + timedelta(seconds=1),
                fiql_filter=search_filter,
            )

I'm guessing you want to widen seconds with "i"?

**Comment by bug-or-feature** [comment]
No. You have misunderstood. The workaround is the manual set of steps described [here](https://community.ig.com/forums/topic/23128-how-does-the-relation-between-dealid-and-dealreference-work-in-the-different-api-endpoints/). 
Although this ticket states that I intended to build support for that workaround into the framework, and it references a commit I made on 2024-01-06, I never merged the branch that contained that commit, and the support was never added to the library. In my view it is too complex to add support for this in real time. Do not use the code in that commit, it is not safe. 
If I was to attempt this again I'd look at doing it as a batch process - end of day kind of thing. 
Closing the ticket
